registry/registry.go

```diff
@@ -218,17 +219,24 @@ func (c *client) tagsToRepository(cancel func(), client backend, lookupName, ima
 		err   error
 	}
 
+	toFetch := make(chan string, len(tags))
 	fetched := make(chan result, len(tags))
 
-	for _, tag := range tags {
-		go func(t string) {
-			img, err := c.lookupImage(client, lookupName, imageName, t)
-			if err != nil {
-				c.Logger.Log("registry-metadata-err", err)
+	for i := 0; i < maxConcurrency; i++ {
+		go func() {
+			for tag := range toFetch {
+				img, err := c.lookupImage(client, lookupName, imageName, tag)
+				if err != nil {
+					c.Logger.Log("registry-metadata-err", err)
+				}
+				fetched <- result{img, err}
 			}
-			fetched <- result{img, err}
-		}(tag)
+		}()
+	}
+	for _, tag := range tags {
+		toFetch <- tag
 	}
+	close(toFetch)
```

Comment:
I need to check this, but I think you need to wait here.

I think there's a chance that you will close the channel before the channel has finished emptying.

Either use a `waitgroup.Wait()` or some othe channel to signal that it is finished.